bond_app/authentication.py

```diff
@@ -76,66 +53,23 @@ def auth_user(self, request_state, token_info_fn=_token_info):
             raise exceptions.Unauthorized('Malformed Authorization header, must be in the form of "bearer [token]".')
 
         token = auth_header_parts[1]
-
-        google_user_info = self.cache_api.get(key='access_token:' + token)
-        if google_user_info is None:
-            google_user_info = self._fetch_user_info(token, token_info_fn)
-            # cache for 10 minutes or until token expires
-            expires_in = min([google_user_info.expires_in, self.config.max_token_life])
-            logging.debug("caching token %s for %s seconds", token, expires_in)
-            self.cache_api.add(key='access_token:' + token, value=google_user_info, expires_in=expires_in)
-        else:
-            logging.debug("auth token cache hit for token %s", token)
-
-        sam_user_info = self.cache_api.get(namespace="SamUserInfo", key=google_user_info.id)
+        sam_user_info = None
+
+        # Note: Datastore cache keys have a hardcoded limit of 1500 bytes. JWTs can exceed that, which
+        # unfortunately means we can't cache them.
+        should_cache_token = 1 <= len(token.encode("utf-8")) <= 1500
```

Comment:
I don't know anything about this codebase, but in general I'd expect a check like this to be done inside the API (`self.cache_api`) instead of in the caller.